File: data-jdbc-postgresql/src/main/java/com/example/data/jdbc/postgresql/AuthorRepository.java
package com.example.data.jdbc.postgresql;

import java.util.Optional;

import com.example.data.jdbc.postgresql.model.Author;

import org.springframework.data.jdbc.repository.query.Query;
import org.springframework.data.repository.ListCrudRepository;

public interface AuthorRepository extends ListCrudRepository<Author, Long> {

	Optional<Author> findByNameContainingIgnoreCase(String partialName);

	@Query("SELECT * FROM author a WHERE a.name = :name LIMIT 1")
	Optional<Author> queryFindByName(String name);

}

File: data/data-jdbc-h2-kotlin/src/main/kotlin/com/example/data/jdbc/h2/kotlin/AuthorRepository.kt
package com.example.data.jdbc.h2.kotlin

import com.example.data.jdbc.h2.kotlin.model.Author
import org.springframework.data.jdbc.repository.query.Query
import org.springframework.data.repository.ListCrudRepository
import java.util.*

interface AuthorRepository : ListCrudRepository<Author, Long> {
	fun findByNameContainingIgnoreCase(partialName: String): Author?

	@Query("SELECT * FROM author a WHERE a.name = :name LIMIT 1")
	fun queryFindByName(name: String): Author?
}
